translator/source_code/icone_translator.py

```diff
@@ -353,6 +391,44 @@ def parse_incident(incident):
 
     return feature
 
+#function to validate the incident
+def validate_incident(incident):
+    
+    if not incident or (type(incident) != dict and type(incident) != OrderedDict):
+        logging.warning('incident is empty or has invalid type')
+        return False
+    
+    location = incident.get('location', {})
+    polyline = location.get('polyline')
+    coords = parse_polyline(polyline)
+    street = location.get('street', '')
+
+    starttime = incident.get('starttime')
+    description = incident.get('description')
+    creationtime = incident.get('creationtime')
+    updatetime = incident.get('updatetime')
+    direction = parse_direction_from_street_name(street)
+    if not direction:
+        direction=get_road_direction(coords)
+        if not direction:
+            logging.warning(f'Invalid incident with id = {incident.get("@id")}.unable to parse direction from street name or polyline')
+            return False
+    required_fields = [location, polyline, coords, street, starttime, description, creationtime, updatetime, direction]
+    for field in required_fields:
+        if not field:
+            return False
+            logging.warning(f'Invalid incident with id = {incident.get("@id")}. Not all required fields are present')
```

Comment:
This logging statement never gets hit, can we move above the return line?

Reply:
I moved the logging statement above the _return False_ line.